tests/unit/test_dataapi.py

```diff
@@ -74,13 +76,53 @@ def test_request_iter(self):
     @responses.activate
     def test_results_iter(self):
         """Verify that `result_iter` returns an iterator of results"""
-        for dummy in dummy_responses:
-            responses.add(responses.POST, endpoint, json=dummy)
+        self.dummy_responses[1]['data'] = {'id': 'b'}
+        for dummy in self.dummy_responses:
+            responses.add(responses.POST, self.endpoint, json=dummy)
         client = DataApi()
-        result_iter = client.results_iter(endpoint, security, consumer_secret,
-                                          request, action)
+        result_iter = client.results_iter(self.endpoint, self.security, self.consumer_secret,
+                                          self.request, self.action)
         results = list(result_iter)
 
         assert len(results) == 2
         assert results[0]['id'] == 'a'
         assert results[1]['id'] == 'b'
+
+    @responses.activate
+    def test_results_iter_error_status(self):
+        """Verify that a DataApiException is raised http status is not ok"""
+        for dummy in self.dummy_responses:
+            responses.add(responses.POST, self.endpoint, json={}, status=500)
+        client = DataApi()
+        with self.assertRaises(DataApiException) as cm:
+            list(client.results_iter(self.endpoint, self.security, self.consumer_secret,
+                                     self.request, self.action))
+            
+            self.assertEqual("server returned HTTP status 500", str(cm.exception))
+
+    @responses.activate
+    def test_results_iter_no_meta_status(self):
+        """Verify that a DataApiException is raised when 'meta' 'status' is None"""
+        for response in self.dummy_responses:
+            response['meta']['status'] = None
+
+        for dummy in self.dummy_responses:
+            responses.add(responses.POST, self.endpoint, json=dummy)
+        client = DataApi()
+        with self.assertRaises(DataApiException) as cm:
+            list(client.results_iter(self.endpoint, self.security, self.consumer_secret,
+                                     self.request, self.action))
+            
+            self.assertEqual("server returned unsuccessful status:", str(cm.exception))
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously. -->
**🎯 suggestion (non-blocking):** maybe `assertRaisesRegex` could be useful here to perform the message checking in the context block?

Ref: https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertRaisesRegex

`pytest` also has similar functionality that maybe we could use in the future 😄 

```suggestion
        with self.assertRaises(DataApiException, "server returned unsuccessful status"):
            list(client.results_iter(self.endpoint, self.security, self.consumer_secret,
                                        self.request, self.action))
```

Reply:
Love it! Absolutely